base/MAPL_ObsUtil.F90

```diff
@@ -913,4 +953,27 @@ subroutine test_conversion
 
   end subroutine test_conversion
 
+
+  subroutine fglob(search_name, filename, rc)     ! give the last name
+    character(len=*), intent(in) ::  search_name
+    character(len=*), intent(INOUT) :: filename
+    integer, optional, intent(out)  :: rc
+
+    character(kind=C_CHAR, len=:), allocatable :: c_search_name
+    character(kind=C_CHAR, len=512) :: c_filename
+    integer n, status, slen
+
+    n=len(trim(search_name))
+    allocate(character(kind=C_CHAR,len=n+1) :: c_search_name)
+    c_search_name(1:n)=search_name(1:n)
+    c_search_name(n+1:n+1)=c_null_char
```

Comment:
```suggestion
    c_search_name = trim(search_name) // C_NULL_CHAR
```

Reply:
Thanks! This helps!  I have changed the code.